src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,10 +1753,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
```

Comment:
MRRT is multi resource refresh token. 
My understanding is, you shouldn't remove other app's entry. It also depends on the detailed error code server return to us, if they explicitly tell us refresh token request fails because token is expired, then we probably should consider for this. Ryan and I talked, there should be further improvement on this, we can probably have a timestamp  on the token cache item, when we try to find family item, we usually find the latest one.